inference-engine/src/mkldnn_plugin/memory_desc/dnnl_blocked_memory_desc.cpp

```diff
@@ -0,0 +1,801 @@
+    }
+    std::vector<size_t> outer_block_dims(std::begin(dims), std::begin(dims) + outer_ndims);
+    for (size_t i = 0; i < outer_block_dims.size(); i++) {
+        outer_block_dims[i] = div_up(outer_block_dims[i], total_block_per_dim[i]);
+    }
+
+    // order of outer dims. In case of IOhw_ will be {1, 0, 2, 3}
+    std::vector<size_t> outer_order(outer_ndims);
+    std::iota(outer_order.begin(), outer_order.end(), 0);
+    std::sort(outer_order.begin(), outer_order.end(),
+              [&blk_desc, &outer_block_dims](size_t ind_l, size_t ind_r) {
+                  return (blk_desc.strides[ind_l] > blk_desc.strides[ind_r]) ||
+                         (blk_desc.strides[ind_l] == blk_desc.strides[ind_r] && outer_block_dims[ind_l] > outer_block_dims[ind_r]);
+              });
+
+
+    // blocked order
+    // [new_outer_order] U [inner_idxs]
+    SizeVector blk_order(total_ndims, 0);
+    std::copy(outer_order.begin(), outer_order.end(), blk_order.begin());
+    std::copy(blk_desc.inner_idxs, blk_desc.inner_idxs + blk_desc.inner_nblks, blk_order.begin() + dims.size());
+    order.swap(blk_order);
+}
+
+bool DnnlBlockedMemoryDesc::hasLayoutType(LayoutType layoutType) const {
+    switch (layoutType) {
+        case LayoutType::ncsp:
+            return isPlainFormat();
+        case LayoutType::nspc:
+            return isTailCFormat();
+        case LayoutType::nCsp8c:
+            return isBlockedCFormat(8);
+        case LayoutType::nCsp16c:
+            return isBlockedCFormat(16);
+        default:
+            return false;
+    }
+}
+
+bool DnnlBlockedMemoryDesc::isPlainFormat() const {
+    if (shape.getRank() != order.size()) {
+        return false;
+    }
+    for (size_t i = 0; i < order.size(); ++i) {
+        if (order[i] != i) {
+            return false;
+        }
+    }
+    return true;
+}
+
+bool DnnlBlockedMemoryDesc::isBlockedCFormat(size_t blk_size) const {
+    const auto &blocking = desc.data.format_desc.blocking;
+
+    if (desc.data.format_kind !=dnnl_blocked ||
+        blocking.inner_nblks != 1 ||
+        blocking.inner_idxs[0] != 1)
+        return false;
+
+    if ((order.size() - shape.getRank()) != 1) {
+        return false;
+    }
+    for (size_t i = 0; i < order.size() - 1; ++i) {
+        if (order[i] != i) {
+            return false;
+        }
+    }
+    if (blk_size != UNREACHABLE_DIM && blk_size != blocking.inner_blks[0]) {
+            return false;
+    }
+
+    return true;
+}
+
+bool DnnlBlockedMemoryDesc::isTailCFormat() const {
+    if (shape.getRank() < 3) {
+        return false;
+    }
+    if (shape.getRank() != order.size()) {
+        return false;
+    }
+    if (!std::is_sorted(order.begin(), --order.end())) {
+        return false;
+    }
+    if (order.back() != 1) {
+        return false;
+    }
+    return true;
+}
+
+std::unique_ptr<MemoryDesc> DnnlBlockedMemoryDesc::cloneWithNewDimsImp(const VectorDims &dims) const {
```

Comment:
Please plase TODO about non default strides handling. The same as for CpuBlockedDesc